In [1]:
import pandas as pd
import numpy as np

url = 'playstore_limpio.csv'
df = pd.read_csv(url)

----

### Reto 1: La Trampa de las Unidades de Medida

In [15]:
def convertir_size(size):
    if pd.isna(size) or size == "Varies with device":
        return np.nan
    elif isinstance(size, (int, float)):
        return float(size)
    elif size.endswith("M"):
        return float(size[:-1])
    elif size.endswith("k"):
        return float(size[:-1]) / 1024
    return np.nan

df['Size'] = df['Size'].apply(convertir_size)

promedio = df['Size'].mean()

print("Promedio de Size:", promedio)

Promedio de Size: 20.41973193945847


### Pregunta a responder: 

Una vez convertida la columna a valores numéricos (Megabytes), ejecuta el método .mean(). ¿Cuál es el peso promedio en Megabytes de las apps en la Play Store?

El peso promedio de las apps en la Play Store es de 20.42 MB



----

### Reto 2: El Tipo de Dato Cronológico

In [17]:
df['Last Updated'] = pd.to_datetime(df['Last Updated'])

df['Year_Updated'] = df['Last Updated'].dt.year

df["Year_Updated"].value_counts()

Year_Updated
2018    6271
2017    1785
2016     779
2015     448
2014     203
2013     108
2012      26
2011      15
2010       1
Name: count, dtype: int64

### Pregunta a responder: 

Utilizando la sumarización categórica (value_counts()) sobre tu nueva columna Year_Updated, ¿en qué año se actualizó la mayor cantidad de aplicaciones en nuestro dataset?

El año 2018 fue cuando se actualizaron mas aplicaciones.



----

## Reto 3: La Decisión Arquitectónica

In [18]:
df["Size"].isna().sum()
df["Size"].median()

np.float64(12.0)

In [21]:
df["Size"] = df["Size"].fillna(df["Size"].median())
print("NaN después de la mediana:", df["Size"].isna().sum())


NaN después de la mediana: 0


### Pregunta a responder: 

Redacten una breve justificación técnica de 3 líneas explicando qué método eligieron y por qué lo consideran superior para no dañar al modelo de predicción.

Elegi imputar los valores faltantes con la mediana para poder conservar la mayor cantidad posible de registros del dataset y para poder evitar perder informacion que podra ser util.